sklearn/preprocessing/imputation.py

```diff
@@ -298,6 +396,80 @@ def _dense_fit(self, X, strategy, missing_values, axis):
 
             return most_frequent
 
+        # KNN
+        elif strategy == "knn":
```

Comment:
I don't think knn is a separate strategy. you could still opt to adopt the median of the k-nearest neighbors' values. (Perhaps this makes limited sense given the masked-euclidean metric, and maybe we should use masked-manhattan (equivalently Gower distance, I think) in the `strategy='median'` case...?)